### PR TITLE
[BEAM-1575] Adding validatesrunner test for metrics in sources

### PR DESCRIPTION
**Pull request by pabloem**

### CODE REVIEW

**Comment by pabloem** [comment]
r: @chamikaramj 
Adding a ValidatesRunner test that checks that  metrics work within sources.

**Comment by pabloem** [comment]
The failure is due to lint failure, but it seems to be due to a different set of files (not affected by my change). I guess this is normal?

File: sdks/python/apache_beam/transforms/ptransform_test.py [hunk 1]
```diff
@@ -169,6 +172,69 @@ def test_do_with_side_outputs_maintains_unique_name(self):
     pipeline.run()
 
   @attr('ValidatesRunner')
+  def test_read_from_text_metrics(self):
+    from apache_beam.io import iobase
+    from apache_beam.io.range_trackers import OffsetRangeTracker
+    class CountingSource(iobase.BoundedSource):
```
Comment (chamikaramj) [inline sdks/python/apache_beam/transforms/ptransform_test.py]: Can you re-use the CountingSource available in following location ?

https://github.com/apache/beam/blob/master/sdks/python/apache_beam/examples/snippets/snippets.py#L606

**Comment by pabloem** [comment]
This failure is from java. The Python tests pass.

**Comment by pabloem** [comment]
@chamikaramj ping? : )

**Comment by chamikaramj** [comment]
Retest this please.

**Comment by pabloem** [comment]
I have tested this on the dataflow service. Also I just relaunched the JEnkins tests.

**Comment by chamikaramj** [comment]
LGTM

File: sdks/python/apache_beam/transforms/ptransform_test.py [hunk 2]
```diff
@@ -169,6 +172,30 @@ def test_do_with_multiple_outputs_maintains_unique_name(self):
     pipeline.run()
 
   @attr('ValidatesRunner')
+  def test_read_from_text_metrics(self):
+    from apache_beam.examples.snippets.snippets import CountingSource
```
Comment (aaltay) [inline sdks/python/apache_beam/transforms/ptransform_test.py]: Why is this import not at the top?

Comment (sb2nov) [inline sdks/python/apache_beam/transforms/ptransform_test.py]: I think examples should be completely isolated and nothing inside actual beam code should use anything from the examples package. We should remove this.

Comment (aaltay) [inline sdks/python/apache_beam/transforms/ptransform_test.py]: Agreed. This is not clean. The solution here is to copy `CountingSource` or maybe make `CountingSource` a first class test class in the SDK. Could you file a JIRA issue for this?

Comment (sb2nov) [inline sdks/python/apache_beam/transforms/ptransform_test.py]: Done, created: https://issues.apache.org/jira/browse/BEAM-2163

**Pull request closed without merging**